libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKResourceUtils.m

```diff
@@ -54,8 +54,15 @@ + (UIImage *)imageNamed:(NSString *)name
 {
     NSBundle *bundle = [NSBundle bundleForClass:[self class]];
     NSAssert(name != nil, @"name must contain a value.");
+    UIImage *image = [UIImage imageNamed:name inBundle:bundle compatibleWithTraitCollection:nil];
+    if (image) {
+        return image;
+    }
     
-    return [UIImage imageNamed:name inBundle:bundle compatibleWithTraitCollection:nil];
+    // get from main bundle
+    bundle = [NSBundle mainBundle];
```

Comment:
@khawkins, this is because a bug we ran into when consuming Salesforce Mobile SDK using SalesforceKit instead. (which is what Service SDK does). Currently SalesforceKit does not contain images & localized strings properly and Kunal will fix it for 4.2 In between, we need to fix it so that when SalesforceKit is consumed, it can use images and localized string from main bundle as back up